chasm/context.go

```diff
@@ -11,7 +13,8 @@ type Context interface {
 
 	// NOTE: component created in the current transaction won't have a ref
 	// this is a Ref to the component state at the start of the transition
-	Ref(Component) (ComponentRef, bool)
+	RefC(Component) (ComponentRef, bool)
+	RefD(proto.Message) (ComponentRef, bool)
```

Comment:
I don't think we want to expose the Ref to a Data, for now at least. Component is the most basic unit someone outside the CHASM framework can interact with. 
We probably need a different type of Ref (instead of ComponentRef) if we want to make Data node directly addressable from outside in the future. 

No concern on making this a private method and use it for DataPointer implementation and only expose Ref(Component)